TombEditor/EditorActions.cs

```diff
@@ -1995,78 +2058,36 @@ public static void TexturizeGroup(Room room, SectorSelection selection, SectorSe
-                                break;
-
-                            case BlockFace.NegativeZ_Middle:
-                                TexturizeWallSection(room, new VectorInt2(x, z), Direction.NegativeZ, BlockFaceType.Wall, texture, xSubs, iterX, unifyHeight ? GetAreaExtremums(room, area, Direction.NegativeZ, BlockFaceType.Wall) : null);
-                                break;
-
-                            case BlockFace.NegativeZ_RF:
-                            case BlockFace.NegativeZ_WS:
-                                TexturizeWallSection(room, new VectorInt2(x, z), Direction.NegativeZ, BlockFaceType.Ceiling, texture, xSubs, iterX, unifyHeight ? GetAreaExtremums(room, area, Direction.NegativeZ, BlockFaceType.Ceiling) : null);
-                                break;
-
-                            case BlockFace.PositiveZ_QA:
-                            case BlockFace.PositiveZ_ED:
-                                TexturizeWallSection(room, new VectorInt2(x, z), Direction.PositiveZ, BlockFaceType.Floor, texture, xSubs, iterX, unifyHeight ? GetAreaExtremums(room, area, Direction.PositiveZ, BlockFaceType.Floor) : null);
-                                break;
-
-                            case BlockFace.PositiveZ_Middle:
-                                TexturizeWallSection(room, new VectorInt2(x, z), Direction.PositiveZ, BlockFaceType.Wall, texture, xSubs, iterX, unifyHeight ? GetAreaExtremums(room, area, Direction.PositiveZ, BlockFaceType.Wall) : null);
-                                break;
-
-                            case BlockFace.PositiveZ_RF:
-                            case BlockFace.PositiveZ_WS:
-                                TexturizeWallSection(room, new VectorInt2(x, z), Direction.PositiveZ, BlockFaceType.Ceiling, texture, xSubs, iterX, unifyHeight ? GetAreaExtremums(room, area, Direction.PositiveZ, BlockFaceType.Ceiling) : null);
-                                break;
-
-                            case BlockFace.DiagonalQA:
-                            case BlockFace.DiagonalED:
-                                TexturizeWallSection(room, new VectorInt2(x, z), Direction.Diagonal, BlockFaceType.Floor, texture);
-                                break;
-
-                            case BlockFace.DiagonalMiddle:
-                                TexturizeWallSection(room, new VectorInt2(x, z), Direction.Diagonal, BlockFaceType.Wall, texture);
-                                break;
-
-                            case BlockFace.DiagonalRF:
-                            case BlockFace.DiagonalWS:
-                                TexturizeWallSection(room, new VectorInt2(x, z), Direction.Diagonal, BlockFaceType.Ceiling, texture);
-                                break;
-                        }
+                        if (pickedFace.IsFloorWall() && pickedFace.IsNegativeX())
```

Comment:
This code is extremely hard to read. Is there any way you can convert it to a case switch?

Reply:
No, because of how many new enum cases there are, it would increase line count for no reason and this is more flexible.